SplashScreen/app/src/main/java/com/example/android/splashscreen/MainViewModel.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.splashscreen
+
+import android.app.Application
+import android.app.UiModeManager
+import android.content.Context
+import androidx.core.content.edit
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+private const val PREFS_MAIN = "main"
+private const val PREF_NIGHT_MODE = "night_mode"
+
+class MainViewModel(application: Application) : AndroidViewModel(application) {
+
+    // SharedPreferences for saving theme settings. Consider using
+    // [DataStore][https://developer.android.com/topic/libraries/architecture/datastore] or
+    // [Room][https://developer.android.com/training/data-storage/room] in real-life apps.
+    private val prefs = application.getSharedPreferences(PREFS_MAIN, Context.MODE_PRIVATE)
+
+    /**
+     * The current night mode setting saved and persisted by the system.
+     */
+    val nightMode: LiveData<Int>
+        get() = _nightMode
+    private val _nightMode = MutableLiveData<Int>()
+
+    /**
+     * True when the initial data in ViewModel is ready, and the app is ready to draw its main
+     * content.
+     */
+    var isReady = false
+        private set
+
+    init {
+        viewModelScope.launch {
+            val nightMode = withContext(Dispatchers.IO) {
+                prefs.getInt(PREF_NIGHT_MODE, UiModeManager.MODE_NIGHT_AUTO)
```

Comment:
I think `getSharedPreferences()` should probably be done in the bg too

Reply:
Done. It is now in a background thread.